pkg/client/testing/core/fixture.go

```diff
@@ -57,7 +57,10 @@ type ObjectScheme interface {
 // TODO: add support for sub resources
 func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
 	return func(action Action) (bool, runtime.Object, error) {
-		kind, err := mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetResource()})
+		resource := action.GetResource()
+		kind, err := mapper.KindFor(resource)
+		// This is a temporary fix. Because there is no internal resource, so the caller has no way to express that it expects to get an internal kind back. A more proper fix will be directly specify the Kind when build the action.
```

Comment:
I added a sentence on what's the proper fix in my mind, but I don't think we'll have time to implement it any soon.